04-onion-routing.md

```diff
@@ -1407,6 +1438,117 @@ The _origin node_:
+which capture the fees that it paid and the opportunity cost that holding it 
+for `resolution_time` incurred. 
+- `fees`: the fees paid by a forwarded HTLC (as described in [BOLT #7](07-routing-gossip.md#htlc-fees), 
+  equal to 0 if the HTLC was not fulfilled).
+- `opportunity_cost`: `ceil ( (resolution_time - resolution_period) / resolution_period) * fees`
+
+Given that `resolution_time` will be > 0 in practice, `opportunity_cost` is 0 
+for HTLCs that resolve within `resolution_period`, and charges the `fees` that 
+the HTLC would have earned per period it is held thereafter. This cost accounts 
+for the slot and liquidity that could have otherwise been paid for by 
+successful, fast resolving HTLCs during the `resolution_time` the HTLC was 
+locked in the channel.
+
+For every resolved incoming HLTC a peer has forwarded through a node, its 
+`effective_fees` are calculated as follows:
+- if `endorsed` = 1 in `update_add_htlc`:
+  - `effective_fees` = `fees` - `opportunity_cost` 
+- otherwise: 
+  - if successfully resolved AND `resolution_time` <= `resolution_period`
+    - `effective_fees` = `fees` * 0.5
+  - otherwise: 
+    - `effective_fees` = 0
+
+Incoming in-flight HTLCs have a negative impact on reputation, as their 
+influence is unknown until time of resolution. The `outstanding_risk` of each 
+in flight HTLC is calculated as follows: 
+- if `endorsed` = 1 in `update_add_htlc`:
+  - `outstanding_risk` = `fees` * ( `cltv_expiry` - `height_added` * 10 * 60), 
+    where `height_added` is the block height at which the HLTC was irrevocably
+    committed to by the receiving node.
+
+The `effective_fees` that a peer has paid our node are compared to our total 
+routing revenue to classify a peer's reputation as "good" or "neutral". This
+relates the fees that must be paid to earn "good" reputation to the damage that 
+can be done by abusing it.
+
+Routing revenue is assessed over a sliding window, so that reputation 
+classification is related to the node's current routing activity. Nodes MAY
+choose a window per their preferences - shorter windows being more reactive to 
+recent routing patterns, and longer windows aggregating trends over time.
+
+- `routing_window`: the period of time in which total routing revenue is 
+  assessed (default: 10 minutes * the maximum number of blocks a HTLC may be
+  held before the node will send `expiry_too_far`, as outlined in [Failure Messages](#failure-messages)).
+- `routing_revenue`: the sum of all fees paid to the node to forward HTLCs
+  over the interval [ `now` - `routing_window` ; `now` ].
+
+The total `effective_fees` that an individual peer has generated are assessed
+over a longer period of time to relate its reputation classification to the 
+routing activity of the node (represented by `routing_revenue`).
+
+- `reputation_window`: the period of time over which the node's `effective_fees`
+  are calculated to classify its reputation (recommended default = 10 * `routing_window`).
```

Comment:
Does this mean that if you're jammed, at maximum ten of your peers can forward through you?

Reply:
This expresses the length of the windows over which you calculate
a) The reputation a peer has accumulated, which is over the longer `reputation_window` period
b) The overall routing revenue of your node, over the shorter `routing_window` period

These values are compared in order to decide whether the accumulated reputation overcomes the accumulated routing revenue, in order to grant or not the `endorsed` flag.

> Does this mean that if you're jammed, at maximum ten of your peers can forward through you?

How did you come to this conclusion?

Reply:
think it's a misunderstanding